protocol-units/da-sequencer/node/src/storage/mod.rs

```diff
@@ -1,36 +1,124 @@
-use crate::batch::DaBatch;
-use crate::batch::FullnodeTx;
-use crate::block::BlockHeight;
-use crate::block::SequencerBlock;
-use crate::block::SequencerBlockDigest;
-use crate::celestia::CelestiaHeight;
-use crate::error::DaSequencerError;
+use crate::{
+	batch::{DaBatch, FullNodeTx},
+	block::{BlockHeight, SequencerBlock, SequencerBlockDigest},
+	celestia::CelestiaHeight,
+	error::DaSequencerError,
+};
+use bincode;
+use movement_types::block::{Block, BlockMetadata, Id};
+use movement_types::transaction::Transaction;
+use rocksdb::{ColumnFamilyDescriptor, Options, WriteBatch, DB};
+use std::{collections::BTreeSet, result::Result, sync::Arc};
 
-pub struct Storage {}
+pub mod cf {
+	pub const PENDING_TRANSACTIONS: &str = "pending_transactions";
+	pub const BLOCKS: &str = "blocks";
+	pub const BLOCKS_BY_DIGEST: &str = "blocks_by_digest";
+}
+
+#[derive(Debug)]
+pub struct Storage {
+	db: Arc<DB>,
+}
 
 impl Storage {
+	pub fn try_new(path: &str) -> Result<Self, DaSequencerError> {
+		let mut options = Options::default();
+
+		options.create_if_missing(true);
+		options.create_missing_column_families(true);
+
+		let pending_transactions_cf =
+			ColumnFamilyDescriptor::new(cf::PENDING_TRANSACTIONS, Options::default());
+		let blocks_cf = ColumnFamilyDescriptor::new(cf::BLOCKS, Options::default());
+		let blocks_by_digest_cf =
+			ColumnFamilyDescriptor::new(cf::BLOCKS_BY_DIGEST, Options::default());
+
+		let db = DB::open_cf_descriptors(
+			&options,
+			path,
+			[pending_transactions_cf, blocks_cf, blocks_by_digest_cf],
+		)
+		.map_err(|e| DaSequencerError::Generic(e.to_string()))?;
+
+		Ok(Storage { db: Arc::new(db) })
+	}
+
 	/// Save all batch's Tx in the pending Tx table. The batch's Tx has been verified and validated.
 	pub fn write_batch(
 		&self,
-		batch: DaBatch<FullnodeTx>,
+		batch: DaBatch<FullNodeTx>,
 	) -> std::result::Result<(), DaSequencerError> {
-		todo!();
+		let cf = self.db.cf_handle(cf::PENDING_TRANSACTIONS).ok_or_else(|| {
+			DaSequencerError::Generic("Missing column family: pending_transactions".into())
+		})?;
+		let tx = batch.data;
+		let key = tx.id.0;
+		let value = bincode::serialize(&tx)
+			.map_err(|e| DaSequencerError::Generic(format!("Serialization error: {}", e)))?;
+
+		let mut write_batch = WriteBatch::default();
+		write_batch.put_cf(&cf, key, value);
+
+		self.db
+			.write(write_batch)
+			.map_err(|e| DaSequencerError::Generic(format!("DB write error: {}", e)))?;
```

Comment:
You're writing a batch in your user-defined format, but there is no need to use a RocksDB batch to write a single KV pair.